src/Adapter/MSTest.TestAdapter/Execution/ExceptionHelper.cs

```diff
@@ -77,18 +77,22 @@ internal static class ExceptionHelper
         bool first = true;
         while (stackTraces.Count != 0)
         {
-            result.AppendFormat(
-                    CultureInfo.CurrentCulture,
-                    "{0} {1}{2}",
-                    first ? string.Empty : (Resource.UTA_EndOfInnerExceptionTrace + Environment.NewLine),
-                    stackTraces.Pop(),
-                    Environment.NewLine);
+            if (!first)
+            {
+                result.Append(utaEndOfInnerExceptionTrace);
+            }
+
+            result.Append(' ');
+            result.AppendLine(stackTraces.Pop());
+
             first = false;
         }
 
         return CreateStackTraceInformation(ex, true, result.ToString());
     }
 
+    private static readonly string utaEndOfInnerExceptionTrace = Resource.UTA_EndOfInnerExceptionTrace + Environment.NewLine;
```

Comment:
Note: Caching will change behavior.

Previously, CurrentUICulture was respected every time `Resource.UTA_EndOfInnerExceptionTrace` is accessed.